src/util/io.rs

```diff
@@ -63,11 +68,105 @@ impl Read for FileChunk {
   }
 }
 
+/// Positional write trait.
+///
+/// Tracks the current position in the stream, should be used together with `Write`.
+/// This does not require &mut self to get the position.
+pub trait Position {
+  /// Returns the current position in the stream.
+  fn pos(&self) -> u64;
+}
+
+/// Struct that represents `File` output stream with position tracking.
+/// Used as a sink in file writer.
+pub struct FileSink {
```

Comment:
Since we named this `FileSink`, should we rename `FileChunk` to `FileSource`?

Reply:
Yep, sounds good - will update.